### DISCUSSION TITLE
How to enable clipboard when using HTTP?

### DISCUSSION THREAD

**Discussion by 0x915** [coder/code-server#6603] [Q&A]
`code-server is being accessed in an insecure context. Web views, the clipboard, and other functionality may not work as expected.`

I have code-server deployed on a LAN and will only access it via local IP. I am unable to provide a generic https access environment as there are no certificates. Is there any way to make code-server work in normal mode with http?

**Comment by code-asher** [comment]
No, HTTPS is required for those things to work.  This is a browser requirement, unfortunately, nothing we can do about it from code-server's end.

**Reply by code-asher** [reply]
Oh but you could possibly workaround it in your browser.  I believe there is a setting to allow certain domains, so you could add your IP.  You could also use a self-signed certificate and add it to the browser or system trust store.

**Reply by code-asher** [reply]
In Chrome for example check out `chrome://flags/#unsafely-treat-insecure-origin-as-secure`.
